File: java/dsa/data-structures/graph/graph.java
void dfs(String nodeId, Set<String> visited) { // @viz:def
    visited.add(nodeId); // @viz:mark
    process(nodeId); // @viz:visit

    for (String neighbor : getNeighbors(nodeId)) { // @viz:for
        if (!visited.contains(neighbor)) { // @viz:check
            dfs(neighbor, visited); // @viz:dfs
        }
    }
}

File: python/dsa/data-structures/graph/graph.py
def dfs(node_id, visited=None): # @viz:def
    if visited is None: visited = set()
    visited.add(node_id) # @viz:mark
    process(node_id) # @viz:visit
    for neighbor in get_neighbors(node_id): # @viz:for
        if neighbor not in visited: # @viz:check
            dfs(neighbor, visited) # @viz:dfs